src/java.base/windows/classes/sun/nio/fs/WindowsFileCopy.java

```diff
@@ -174,7 +177,13 @@ static void copy(final WindowsPath source,
 
         // Use CopyFileEx if the file is not a directory or junction
         if (!sourceAttrs.isDirectory() && !sourceAttrs.isDirectoryLink()) {
-            final int flags = (!followLinks) ? COPY_FILE_COPY_SYMLINK : 0;
+            long size = 0;
+            try {
+                size = Files.size(source);
+            } catch (IOException ignored) {
+            }
+            final int flags = ((!followLinks) ? COPY_FILE_COPY_SYMLINK : 0) |
+                ((size > UNBUFFERED_IO_THRESHOLD) ? COPY_FILE_NO_BUFFERING : 0);
```

Comment:
The size of the source file is already in sourceAttrs.size(), no need to call back into the API to access the file again.

Reply:
Fixed.